mm2src/coins/utxo/utxo_builder/utxo_arc_builder.rs

```diff
@@ -245,25 +234,31 @@ async fn block_header_utxo_loop<T: UtxoCommonOps>(
             },
         };
 
+        let total_block_height = match coin.as_ref().rpc_client.get_block_count().compat().await {
```

Comment:
We can request the `block_count` once at the beginning of this loop, and don't request it every iteration until `from_block_height + chunk_size < total_block_height`.

Reply:
done